Handle case of multiple IP range, combination of CIDR and IP range

ShubhamRasal opened the pull request:
- Input may contain the multiple CIDRs and multiple IP ranges
  eg. `166.8.0.0/16,166.11.0.0/16,166.9.0.0-166.10.255.255`
- Handle the above input for arggregation, count, etc
- Move the string split logic from `main()` to `process()` function
- Add the new variable IPRangeMap which stores multiple IP range

ShubhamRasal committed:
Handle case of multiple IP range, combination of CIDR and IP range

- Input may contain the multiple CIDRs and multiple IP ranges
  eg. `166.8.0.0/16,166.11.0.0/16,166.9.0.0-166.10.255.255`
- Handle the above input for arggregation, count, etc
- Move the string split logic from `main()` to `process()` function
- Add the new variable IPRangeMap which stores multiple IP range

ShubhamRasal committed:
Add the comment for ipRange block

jdk2588 commented on cmd/mapcidr/main.go:
```diff
@@ -324,33 +333,27 @@ func process(wg *sync.WaitGroup, chancidr, outputchan chan string) {
-		if err != nil {
-			gologger.Fatal().Msgf("%s\n", err)
-		}
+		for _, ipRange := range ipRangeMap {
```
You are not doing anything with the `key` of the map, so why need a map in first place?

ShubhamRasal replied:
Remove the map with the list

jdk2588 commented on cmd/mapcidr/main.go:
```diff
@@ -289,20 +282,36 @@ func prepareIPsFromCidrFlagList(items []string) []string {
+		ranger       *ipranger.IPRanger
+		err          error
+		hasSort      = options.SortAscending || options.SortDescending
+		ipRangeMap   = make(map[int][]net.IP)
```
Why do we need a map, why can't the array work? Why do you need iprangeCount also?

ShubhamRasal replied:
Removed the map with the list

ShubhamRasal committed:
Replace ipRangeMap to ipRangeList.

- map key is not being used so dropped the map implementation.
Replace ipRangeMap to ipRangeList.

- map key is not being used so dropped the map implementation.
Replace ipRangeMap to ipRangeList.

- map key is not being used so dropped the map implementation.
Replace ipRangeMap to ipRangeList.

- map key is not being used so dropped the map implementation.
Replace ipRangeMap to ipRangeList.

- map key is not being used so dropped the map implementation.
Replace ipRangeMap to ipRangeList.

- map key is not being used so dropped the map implementation.
Replace ipRangeMap to ipRangeList.

- map key is not being used so dropped the map implementation.
Replace ipRangeMap to ipRangeList.

- map key is not being used so dropped the map implementation.
Replace ipRangeMap to ipRangeList.

- map key is not being used so dropped the map implementation.
